Configuration.props

```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project DefaultTargets="Build" ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!-- Note: MUST be imported *after* $(Configuration) is set! -->
```

Comment:
That'll result in obscure/bizarre errors, as a non-existent configuration could be used for building:

    <!-- "bad" .csproj -->
    <Project ...>
      <Import Project="..\..\Configuration.props" />
      <PropertyGroup>
        <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
      </PropertyGroup>
      <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' " />
      <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' " />
    </Project>

With such a "bad" project, `$(Configuration)` *will be set*, causing MSBuild to look for the `ConfigurationNotSetInConfigurationPropsForProject|AnyCPU` configuration, which doesn't exist.

*At best*, it'll still compile but you have no idea where the output file is.

More likely -- if `#if` is used *at all* -- is that the project will fail to build with some "bizarro" error that in no way suggests that the error is "an invalid `$(Configuration)` value was specified, dummy!".